### PR TITLE
HtmlEditor - support editing of tables and cells properties

### PR DESCRIPTION
**Pull request by alexander-kotov-dx**

### CODE REVIEW

File: js/ui/html_editor/modules/toolbar.js [hunk 1]
```diff
@@ -190,7 +192,9 @@ if(Quill) {
                 insertColumnRight: getTableOperationHandler(this.quill, 'insertColumnRight'),
                 deleteColumn: getTableOperationHandler(this.quill, 'deleteColumn'),
                 deleteRow: getTableOperationHandler(this.quill, 'deleteRow'),
-                deleteTable: getTableOperationHandler(this.quill, 'deleteTable')
+                deleteTable: getTableOperationHandler(this.quill, 'deleteTable'),
+                cellProperties: () => { showCellPropertiesForm(this.editorInstance, $(this.quill.getModule('table').getTable()[2].domNode)); },
+                tableProperties: () => { showTablePropertiesForm(this.editorInstance, $(this.quill.getModule('table').getTable()[0].domNode)); }
```
Comment (DokaRus) [inline js/ui/html_editor/modules/toolbar.js]:
```suggestion
                cellProperties: () => {
                    const domNode = getTargetTableNode(this.quill, 'cell');
                    showCellPropertiesForm(this.editorInstance, $(domNode));
            },
                tableProperties: () => {
                    const domNode = getTargetTableNode(this.quill, 'table');
                    showTablePropertiesForm(this.editorInstance, $(domNode));
            }
```

Let's simplify it. Now it looks like ninja code :) 

```js
getTargetTableNode(quill, partName) {
    const [table, , cell] = this.quill.getModule('table').getTable();
    return tablePart === 'table' ? table,domNode : cell.domNode;
}
```

File: js/ui/html_editor/modules/tableOperations.js [hunk 2]
```diff
@@ -1,3 +1,580 @@
...
+            caption: 'Alignment',
+            colCount: 2,
+            items: [
+                {
+                    itemType: 'simple',
+                    label: { text: 'Horizontal' },
+                    template: () => {
+                        const $content = $('<div>');
+                        editorInstance._createComponent($content, ButtonGroup, {
+                            items: [{ value: 'left', icon: 'alignleft' }, { value: 'center', icon: 'aligncenter' }, { value: 'right', icon: 'alignright' }, { value: 'justify', icon: 'alignjustify' }],
```
Comment (DokaRus) [inline js/ui/html_editor/modules/tableOperations.js]: Localization?

File: js/ui/html_editor/modules/tableOperations.js [hunk 3]
```diff
@@ -1,3 +1,580 @@
...
+                        });
+                        return $content;
+                    }
+                }, {
+                    itemType: 'simple',
+                    label: { text: 'Vertical' },
+                    template: () => {
+                        const $content = $('<div>');
+                        editorInstance._createComponent($content, ButtonGroup, {
+                            items: [{ value: 'top', icon: 'verticalaligntop' }, { value: 'middle', icon: 'verticalaligncenter' }, { value: 'bottom', icon: 'verticalalignbottom' }],
```
Comment (DokaRus) [inline js/ui/html_editor/modules/tableOperations.js]: Localization?

File: js/ui/html_editor/modules/tableOperations.js [hunk 4]
```diff
@@ -1,3 +1,580 @@
...
+                },
+                {
+                    dataField: 'verticalPadding',
+                    label: { text: 'Padding Vertical' },
+                    editorOptions: {
+                        placeholder: 'Pixels'
+                    }
+                },
+                {
+                    label: { text: 'Padding Horizontal' },
```
Comment (DokaRus) [inline js/ui/html_editor/modules/tableOperations.js]: Localization?

File: js/ui/html_editor/modules/tableContextMenu.js [hunk 5]
```diff
@@ -118,7 +135,7 @@ if(Quill) {
         }
 
         prepareCleanCallback() {
-            return () => { this._detachEvents(); };
+            return () => { this._detachEvents(); this._popupForm?.dispose(); };
```
Comment (DokaRus) [inline js/ui/html_editor/modules/tableContextMenu.js]:
```suggestion
            return () => {
                this._detachEvents();
                this._popupForm?.dispose();
            };
```

It is worth keeping the code clear. You can skip the executable line when reading

File: js/ui/html_editor/modules/tableContextMenu.js [hunk 6]
```diff
@@ -67,8 +81,10 @@ if(Quill) {
                             { text: 'Delete Row', icon: 'deleterow', onClick: getTableOperationHandler(this.quill, 'deleteRow') },
                             { text: 'Delete Table', icon: 'deletetable', onClick: getTableOperationHandler(this.quill, 'deleteTable') }
                         ]
-                    }
-                ],
+                    },
+                    { text: 'Cell Properties', icon: 'cellproperties', onClick: (e) => { this.showCellProperties(e); } },
```
Comment (DokaRus) [inline js/ui/html_editor/modules/tableContextMenu.js]: Localization

File: js/ui/html_editor/modules/tableOperations.js [hunk 7]
```diff
@@ -1,3 +1,580 @@
...
+import { noop } from '../../../core/utils/common';
+
+import localizationMessage from '../../../localization/message';
+
+const MIN_HEIGHT = 400;
+const BORDER_STYLES = ['none', 'hidden', 'dotted', 'dashed', 'solid', 'double', 'groove', 'ridge', 'inset', 'outset'];
+let formPopup;
+let applyHandler = noop;
+
+const createFormPopup = (editorInstance) => {
```
Comment (DokaRus) [inline js/ui/html_editor/modules/tableOperations.js]: It isn't a Quill module. Let's move it to `js/ui/html_editor/ui/`

**Review by DokaRus — DISMISSED** [review]

**Review by Zedwag — APPROVED** [review]

**Pull request merged**